doc/kb_sdk_edit_module.md

```diff
@@ -112,32 +109,37 @@ The type definition can either be another previously defined type name, a primit
 container type (list, mapping) or a structure.  In this example we define a string named `contigset_id` and a structure 
 named `FilterContigResults` with two integer fields named `contig_count` and `filtered_contig_count`.
 
-We can use any defined types as input/output parameters to functions.  We define functions using the `funcdef` keyword 
-in this syntax:
+You can use any defined types as input/output parameters to functions but it's wise to pass an parameter object. This
+allows additional parameters to be supplied in future implementations without breaking back-comparability. Functions 
+are defined using the `funcdef` keyword in this syntax:
 
-    funcdef method_name([input parameter list]]) returns ([output parameter list]);
+    funcdef method_name(input_parameter_object_type params) returns (output_parameter_object_type output) authentication required;
 
-Optionally, as we have shown in the example, your method can require authentication by adding that declaration at the 
-end of the method.  All methods that run in the Narrative will require authentication because they need to interact with
-a user's workpace).
+All methods that run in the Narrative will require authentication because they need to interact with a user's workspace).
+Your method can require authentication by adding that declaration at the end of the method.
 
-If you will be loading or saving any data from the workspace, make sure you pass in the *workspace_name* in the input 
-parameters.
+If you will be loading or saving any data from the workspace, make sure you accept a workspace reference in the input 
```

Comment:
So it turns out that KBaseReports only accepts workspace_name as a parameter which means that passing only a workspace id causes extra work. I'll open a ticket on this one